front/lib/resources/group_resource.ts

```diff
@@ -177,6 +181,51 @@ export class GroupResource extends BaseResource<GroupModel> {
     return new this(GroupModel, group.get());
   }
 
+  static async addMember(
+    auth: Authenticator,
+    groupId: string,
+    userId: string,
+    transaction?: Transaction
+  ): Promise<void> {
+    const [owner, group, user] = await Promise.all([
+      Promise.resolve(auth.getNonNullableWorkspace()),
+      this.fetchById(auth, groupId),
+      UserResource.fetchById(userId),
+    ]);
+
+    if (!group) {
+      throw new Error("Group not found.");
+    }
+    if (group.type !== "regular") {
+      throw new Error("Cannot add members to non-regular groups.");
+    }
+    if (!user) {
+      throw new Error("User not found.");
+    }
+
+    const workspace = renderLightWorkspaceType({ workspace: owner });
+    const membership =
+      await MembershipResource.getActiveMembershipOfUserInWorkspace({
+        user,
+        workspace,
+        transaction,
+      });
+
+    if (!membership) {
+      throw new Error("User is not a member of the workspace.");
```

Comment:
Can we use a Result<> instead of void and throwing errors?

Reply:
Agreed given that userId can be anything really here. Same for the initial retrieval of `user`?

Reply:
Done! 

[Meta] We need to enforce more guidelines, cause createMembership for the workspace membership is throwing errors and not returning a Result. We would gain some time if we were all aligned on code conventions 🙏🏻